src/backend/templates/schedule.hbs

```diff
@@ -487,33 +488,81 @@
                         {
                           $(this).hide();
-                        }else if(name!=trackName) {
-                          $(this).hide();
                         } else {
                           $(this).show();
-                          flag=1;
+                          isTime = true;
                         }
                     } else {
                       $(this).hide();
                     }
                   });
-                  if(flag===0) {
+                  if(isTime === false) {
                     $(this).hide();
                   } else {
                     $(this).show();
-                      temp=1;
+                      isDay = true;
                   }
                 });
-                if(temp==0) {
+                if(isDay === false) {
                   $(this).hide();
                 } else {
+                  isRoom = false;
                   $(this).show();
                 }
               });
               $('html, body').animate({scrollTop: 0}, 500);
+
+              if(isRoom === true) {
+                roomName = decodeURIComponent(window.location.href.toString().split('#')[1]);
+                roomFilterMode = 1;
+                trackFilterMode = 0;
+
+                $('.room-filter-info').show();
+                $('#RoomCurFilter').text(roomName);
+                $('.track-filter-info').hide();
+
+                $('.day-filter').each(function () {
+                  var isDay = false;
+                  $(this).find('.time-filter').each(function () {
+                    var isTime = false;
+                    $(this).find('.schedule-track').each(function () {
+                      if ($(this).find('.tip-description').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0 ||
+                        $(this).find('.session-speakers-list a p span').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0 ||
+                        $(this).find('.session-speakers-more').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0 ||
+                        $(this).find('.event').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0 ||
+                        $(this).find('.speaker-name').text().toUpperCase().indexOf(filterVal.toUpperCase()) >= 0) {
+                        var name = $(this).find('#location').text();
+                        var id = $(this).attr('id');
+                        if ((mode && local_storage_events[id] !== 1) || (name !== roomName)) {
+                          $(this).hide();
+                        }else {
+                          $(this).show();
+                          isTime = true;
+                        }
+                      } else {
+                        $(this).hide();
+                      }
+                    });
+
+                    if (isTime === false) {
+                      $(this).hide();
```

Comment:
However [some JS experts recommend](https://twitter.com/paul_irish/status/564443848613847040) using classes instead of direct styles.

With a suitable class `.hide { display: none; }`, that could be done like this:

```js
this.toggleClass('hide', isDay);
```

(And current state can be easily tested with `.hasClass('hide')`)